src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/ResultItem.tsx

```diff
@@ -158,20 +158,30 @@ const ResultItem: React.FC<ResultItemProps & AccessibilityProps> = ({
   );
   const fadeInValueRef = useRef(new Animated.Value(0));
 
+  const fadeIn = useCallback(() => {
+    Animated.timing(fadeInValueRef.current, {
+      toValue: 1,
+      duration: 250,
+      useNativeDriver: true,
+    }).start();
+  }, []);
+
   // Dynamically collapse legs to fit horizontally
   useEffect(() => {
     if (legIconsParentWidth && legIconsContentWidth) {
       if (legIconsContentWidth >= legIconsParentWidth) {
-        setNumberOfExpandedLegs((val) => Math.max(val - 1, 1));
+        setNumberOfExpandedLegs((val) => {
+          const newVal = Math.max(val - 1, 1);
+          if (newVal === val) {
+            fadeIn();
```

Comment:
Not sure about this 🤔 
https://stackoverflow.com/questions/71780110/is-it-okey-to-use-side-effects-in-the-usestate-hook-callback